reactive.js

```diff
@@ -0,0 +1,56 @@
+/* reactive engine. the code here combines functions exposed by the
+ * library into a single function to be run. i keep all code depending
+ * on Bacon here, so if somebody will want to stop using Bacon in the
+ * future, it will be easier */
+var lib = require('./lib2');
+var Bacon = require('bacon');
+
+// this is an adapter function. it gets a Pouch changes object and it
+// returns a Bacon stream
+function changesToStream(changes) {
+  return Bacon.fromBinder(function(sink) {
```

Comment:
You would then create multiple streams, one for each event you care about. But if you want to have the errors flow through the one stream everything(changes, config) is combined into, the way you did it is probably the way to go. Im new to this, but pushing everything down one pipe seems like mixing concerns. But then again you probably just use `outgoing.onError(f)` to catch the error. Actually it seems the right way you did it, now that i think more about it :)

It took me a while to understand that you keep state in the Property and how sampledBy works. Its nice, though.